tests/components/test_producer_app.py

```diff
@@ -0,0 +1,260 @@
+            producer_app.handlers.topic_handler, "create_topics"
+        )
+
+        mock_helm_upgrade_install = mocker.patch.object(
+            producer_app.helm_wrapper, "upgrade_install"
+        )
+
+        mock = mocker.MagicMock()
+        mock.attach_mock(mock_create_topics, "mock_create_topics")
+        mock.attach_mock(mock_helm_upgrade_install, "mock_helm_upgrade_install")
+
+        producer_app.deploy(dry_run=True)
+        mock.assert_has_calls(
+            [
+                mocker.call.mock_create_topics(
+                    to_section=producer_app.to, dry_run=True
+                ),
+                mocker.call.mock_helm_upgrade_install(
+                    "example-name",
+                    "bakdata-streams-bootstrap/producer-app",
+                    True,
+                    "test-namespace",
+                    {
+                        "namespace": "test-namespace",
+                        "streams": {
+                            "brokers": "fake-broker:9092",
+                            "outputTopic": "producer-output-topic",
+                        },
+                    },
+                    HelmUpgradeInstallFlags(
+                        force=False,
+                        repo_auth_flags=RepoAuthFlags(
+                            username=None,
+                            password=None,
+                            ca_file=None,
+                            insecure_skip_tls_verify=False,
+                        ),
+                        timeout="5m0s",
+                        version="2.4.2",
+                        wait=True,
+                        wait_for_jobs=False,
+                    ),
+                ),
+            ],
+        )
+
+    def test_destroy(
+        self,
+        config: PipelineConfig,
+        handlers: ComponentHandlers,
+        mocker: MockerFixture,
+    ):
+        producer_app = ProducerApp(
+            handlers=handlers,
+            config=config,
+            **{
+                "type": "producer-app",
+                "name": "example-name",
+                "version": "2.4.2",
+                "app": {
+                    "namespace": "test-namespace",
+                    "streams": {"brokers": "fake-broker:9092"},
+                },
+                "to": {
+                    "topics": {
+                        "${output_topic_name}": TopicConfig(
+                            type=OutputTopicTypes.OUTPUT, partitions_count=10
+                        ),
+                    }
+                },
+            },
+        )
+        producer_app.handlers = MagicMock()
+        mock_helm_uninstall = mocker.patch.object(
+            producer_app.helm_wrapper, "uninstall"
+        )
+
+        producer_app.destroy(dry_run=True, clean=False, delete_outputs=False)
+
+        mock_helm_uninstall.assert_called_once_with(
+            "test-namespace", "example-name", True
+        )
+
+    # TODO: Assert schema deletion
```

Comment:
@disrupted let's do this tomorrow together. For some reason I couldn't access the schema handler mock 🤔

Reply:
@raminqaf yep, feel free to ping me tomorrow